eip_7594/src/error.rs

```diff
@@ -0,0 +1,22 @@
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error(
+        "Custody group count is invalid: {custody_group_count} expected <= {number_of_custody_groups}", 
```

Comment:
We have a convention of starting errors with a lowercase letter, unless it's an acronym or a title. We should stick to it. There are other violations of this in other places as well.